### PR TITLE
Update to Terraform 0.12

### PR DESCRIPTION
**Pull request by ludoo**
This PR updates the module to work with Terraform 0.12.

It also introduces a breaking change in outputs, where outputs for multiple buckets are switched from maps to list, so that hey preserve the same order as the `names` variable that is passed in, to avoid potential issues.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -14,65 +14,77 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.11.0"
-}
-
 locals {
-  prefix = "${var.prefix == "" ? "" : "${var.prefix}-"}${lower(var.location)}"
+  prefix = var.prefix == "" ? "" : join("-", list(var.prefix, ""))
```
Comment (morgante) [inline main.tf]: Is this intentional? The dropping of location looks like a functionality change.

File: main.tf [hunk 2]
```diff
@@ -14,65 +14,77 @@
...
-  storage_class      = "${var.storage_class}"
-  labels             = "${var.labels}"
-
+  count         = length(var.names)
+  name          = "${local.prefix}${lower(element(var.names, count.index))}"
+  project       = var.project_id
+  location      = var.location
+  storage_class = var.storage_class
+  labels        = var.labels
+  force_destroy = lookup(
```
Comment (morgante) [inline main.tf]: This looks like an unrelated change?

File: main.tf (hunk 1 above)
Comment (ludoo) [reply main.tf]: True, reasoning was that it's easy to inject that in prefix if needed. But let's revert, move to 0.12, then maybe pick this up again.

File: main.tf (hunk 2 above)
Comment (ludoo) [reply main.tf]: You mean force destroy? Yep, forgot I had added it. :) I think it's really needed, it can of course go in a separate PR if you prefer.

File: main.tf (hunk 1 above)
Comment (morgante) [reply main.tf]: Agreed, let's do that in a separate PR if it's needed.

File: main.tf (hunk 2 above)
Comment (morgante) [reply main.tf]: Fine to keep it here, just please add a note to the CHANGELOG.

File: main.tf (hunk 1 above)
Comment (ludoo) [reply main.tf]: Reverted and retested, thanks.

File: main.tf (hunk 2 above)
Comment (ludoo) [reply main.tf]: Added note.

**Comment by morgante** [comment]
LGTM, merge and release at will.
